### PR TITLE
prov reader

### PR DESCRIPTION
**Pull request by alan-stokes**
this allows the provenance data to be read but not written to disk in any form. So basically you can still get prov data and query it in ram if needed. 

Also cleaned out a clone in ASB. 

spynnaker8 pr for tests:
https://github.com/SpiNNakerManchester/sPyNNaker8/pull/403

### CODE REVIEW

**Comment by alan-stokes** [comment]
fixes: https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/issues/578#issuecomment-623979416

File: spinn_front_end_common/interface/spinnaker.cfg [hunk 1]
```diff
@@ -162,7 +163,8 @@ boot_connection_port_num = None
 auto_detect_bmp = False
 turn_off_machine = False
 clear_routing_tables = False
-clear_tags = False
+clear_tags = False"when there is nothing left to remove
```
Comment (rowleya) [inline spinn_front_end_common/interface/spinnaker.cfg:166]: Oops?

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 2]
```diff
@@ -1849,6 +1849,33 @@ def _end_of_run_timing(self):
             self._mapping_time, self._dsg_time, self._load_time,
             self._execute_time, self._extraction_time)
 
+    def _sort_out_provenance_writing(self, executor):
```
Comment (rowleya) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py:1852]: Not just _write_provenance?

File: spinn_front_end_common/interface/spinnaker.cfg (hunk 1 above)
Comment (alan-stokes) [reply spinn_front_end_common/interface/spinnaker.cfg:166]: ..................... how did that get in............... oh well. yep. oops

Comment (alan-stokes) [reply spinn_front_end_common/interface/spinnaker.cfg]: fixed

File: spinn_front_end_common/interface/abstract_spinnaker_base.py (hunk 2 above)
Comment (alan-stokes) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py:1852]: well coz there already is a _write_provenance that that method calls. lol. But will look into merging the two

Comment (alan-stokes) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py:1852]: nah, it makes _write_provenance over more than 1 screen worth of a method

ill rename to gather_provenance_for_writing

**Comment by rowleya** [comment]
Looks fine, but integration tests are now failing (and quite badly)

**Comment by alan-stokes** [comment]
i had not got around to resetting the spynnaker8 tests after the oops. which is what the tests are failing from. its now been reset

**Comment by rowleya** [comment]
All passing now

**Pull request merged**